app/src/main/java/com/android/sample/model/map/LocationViewModel.kt

```diff
@@ -55,4 +60,51 @@ constructor(
       // Handle the permission denial as appropriate
     }
   }
+
+  /**
+   * Calculate the distance between the current location and the given activity location.
+   *
+   * @param activityLocation The location of the activity.
+   * @return The distance in kilometers.
+   */
+  fun getDistanceFromCurrentLocation(activityLocation: Location?): Float? {
```

Comment:
Okay, I understand now your null handling. So you are returning null if one of the two input locations is null. I guess this is useful in the code, as the user may not disclose their location, or as some activities do not have a location. In this case your code will return null, which you should correctly manage in the UI.